targets/csharp/source/source/PlayFabGameserverFileLogger.cs

```diff
@@ -0,0 +1,82 @@
+﻿// GSDK Only has the following supported frameworks for now
+#if NETSTANDARD2_0 || NETCOREAPP2_1
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace PlayFab
+{
+    public interface ILogger
+    {
+        void Start();
+        void Stop();
+        void Log(string message);
+    }
+
+    public static class LoggerFactory
+    {
+        public static ILogger CreateInstance(string logFolder)
+        {
+            return new FilesystemLogger(logFolder);
+        }
+    }
+
+    public class FilesystemLogger : ILogger
+    {
+        private StreamWriter _logWriter;
+        private string _logFolder;
+
+        public FilesystemLogger(string logFolder)
+        {
+            _logFolder = logFolder;
+        }
+
+        public void Log(string message)
+        {
+            if (_logWriter == null)
+            {
+                return; // Logging is disabled
+            }
+
+            _logWriter.WriteLine($"{DateTime.UtcNow.ToString("o")}\t{message}");
```

Comment:
Optional: You can use ?.WriteLine and ?.Flush() and get rid of the null check

Reply:
I think that one I'll keep, if we're ever debugging why logging is disabled, might be useful to have a separate if condition we can add a breakpoint to or such

---
In reply to: [255760674](https://github.com/PlayFab/SDKGenerator/pull/398#discussion_r255760674) [](ancestors = 255760674)